miner/scroll_worker.go

```diff
@@ -453,165 +544,198 @@ func (w *worker) startNewPipeline(timestamp int64) {
 
-	if w.prioritizedTx != nil && w.currentPipeline.Header.Number.Uint64() > w.prioritizedTx.blockNumber {
+	signer := types.MakeSigner(w.chainConfig, w.current.header.Number)
+	if w.prioritizedTx != nil && w.current.header.Number.Uint64() > w.prioritizedTx.blockNumber {
 		w.prioritizedTx = nil
 	}
 	if w.prioritizedTx != nil {
 		from, _ := types.Sender(signer, w.prioritizedTx.tx) // error already checked before
 		txList := map[common.Address]types.Transactions{from: []*types.Transaction{w.prioritizedTx.tx}}
-		txs := types.NewTransactionsByPriceAndNonce(signer, txList, header.BaseFee)
-		if result := w.currentPipeline.TryPushTxns(txs, w.onTxFailingInPipeline); result != nil {
-			w.handlePipelineResult(result)
-			return
+		txs := types.NewTransactionsByPriceAndNonce(signer, txList, w.current.header.BaseFee)
+
+		if shouldCommit, err := w.processTxns(txs); err != nil {
+			return false, fmt.Errorf("failed to include prioritized tx: %w", err)
+		} else if shouldCommit {
+			return true, nil
 		}
 	}
 
 	if len(localTxs) > 0 {
-		txs := types.NewTransactionsByPriceAndNonce(signer, localTxs, header.BaseFee)
-		if result := w.currentPipeline.TryPushTxns(txs, w.onTxFailingInPipeline); result != nil {
-			w.handlePipelineResult(result)
-			return
+		txs := types.NewTransactionsByPriceAndNonce(signer, localTxs, w.current.header.BaseFee)
+		if shouldCommit, err := w.processTxns(txs); err != nil {
+			return false, fmt.Errorf("failed to include locals: %w", err)
+		} else if shouldCommit {
+			return true, nil
 		}
 	}
+
 	if len(remoteTxs) > 0 {
-		txs := types.NewTransactionsByPriceAndNonce(signer, remoteTxs, header.BaseFee)
-		if result := w.currentPipeline.TryPushTxns(txs, w.onTxFailingInPipeline); result != nil {
-			w.handlePipelineResult(result)
-			return
+		txs := types.NewTransactionsByPriceAndNonce(signer, remoteTxs, w.current.header.BaseFee)
+		if shouldCommit, err := w.processTxns(txs); err != nil {
+			return false, fmt.Errorf("failed to include remotes: %w", err)
+		} else if shouldCommit {
+			return true, nil
 		}
 	}
 
-	// pipelineCCC was nil, so the block was built for RPC purposes only. Stop the pipeline immediately
-	// and update the pending block.
-	if pipelineCCC == nil {
-		w.currentPipeline.Stop()
+	return false, nil
+}
+
+// processTxnSlice
+func (w *worker) processTxnSlice(txns types.Transactions) (bool, error) {
+	txsMap := make(map[common.Address]types.Transactions)
+	signer := types.MakeSigner(w.chainConfig, w.current.header.Number)
+	for _, tx := range txns {
+		acc, _ := types.Sender(signer, tx)
+		txsMap[acc] = append(txsMap[acc], tx)
 	}
+	txset := types.NewTransactionsByPriceAndNonce(signer, txsMap, w.current.header.BaseFee)
+	return w.processTxns(txset)
 }
 
-func (w *worker) handlePipelineResult(res *pipeline.Result) error {
-	startingHeader := w.currentPipeline.Header
-	w.currentPipeline.Release()
-	w.currentPipeline = nil
+// processReorgedTxns
+func (w *worker) processReorgedTxns(reason error) (bool, error) {
```

Comment:
Will this process a single block? Or all subsequent blocks too?

Reply:
This handles a single block only. `handleReorg` is responsible of reorging the entire faulty part of the chain.